### ISSUE TITLE
[Question] How do I mark a command as deprecated?

### ISSUE THREAD

**Issue by ryandberger** [yargs/yargs#2115]
I see from https://github.com/yargs/yargs/pull/1559 that it is now possible to mark options as deprecated. I have attempted to use this functionality at a higher level to mark an entire command as deprecated, but it does not work. Is this feature not yet implemented or am I missing something?

This is the usage I am attempting:

```
module.exports = {
  command: 'old [path]',
  deprecated: '[deprecated: use new instead]'
};
```

**Comment by jly36963** [comment]
I believe https://github.com/yargs/yargs/pull/1559 only applies to options, whereas https://github.com/yargs/yargs/pull/1624 is for commands.

## Example

You can use `deprecated: true` or `deprecated: "explanation here"` for both options and commands.

My example uses the `yargs.command(module)` syntax, but it should also work for `yargs.command(cmd, description, builder, handler, commandMiddleware, deprecated)`.

```js
let input

/*
input = '--help'  // you will see `[deprecated]` next to the command "cmd1"
input = 'cmd1 --help' // you will see `[deprecated]` next to the option "deprecated-arg"
*/

yargs(input)
  .command({
    command: 'cmd1',
    desc: 'cmd1 desc',
    builder: yargs => yargs
      .option('arg', {
        type: 'string',
        desc: 'arg desc',
      })
      .option('deprecated-arg', {
        type: 'string',
        desc: 'deprecated-arg desc',
        deprecated: true // deprecate option
        // deprecated: 'use "arg" option instead' // deprecate option (with explanation)
      }),
    handler: argv => {
      console.log(argv)
    },
    deprecated: true // deprecate command
    // deprecated: 'use "cmd2" command instead' // deprecate command (with explanation)
  })
  .command({
    command: 'cmd2',
    desc: 'cmd2 desc',
    builder: yargs => yargs
      .option('arg', {
        type: 'string',
        desc: 'arg desc',
      }),
    handler: argv => {
      console.log(argv)
    },
  })
  .strict()
  .parse()
```

Does this answer your question?

**Comment by ryandberger** [comment]
Yes, I do see the `[deprecated]` indicator next to command `cmd1` when calling `--help` generally. But when I call `cmd1 --help` I don't see the message after the command description which I think is unexpected.

**Comment by ryandberger** [comment]
Would love to see a warning emitted when the command is used. I think that would be particularly useful for automation processes. For your approach, I personally would lean towards option 2 since I imagine developers would want consistent behavior across their app. I think it would also make option 3 more natural to implement when the time comes. Appreciate your help on this!

**Comment by ryandberger** [comment]
Related to this: I noticed that `aliases` attribute also only displays at the top level. Would be helpful to also see '[aliases]` indicator display when calling `cmd1 --help`. Figured it might be convenient to bundle the work together if you agree with my request.

**Comment by AgentEnder** [comment]
@shadowspawn do you know if this was ever decided on further? I'd love to have this, even if its something opt-in like with "yargs().showDeprecationMessages()" or similar.

**Comment by shadowspawn** [comment]
There have not been any further discussions about runtime warnings (that I have seen).

There are a couple of related issues on the help and docs: #2248 #2246
